Move tk to host so that its export is picked up

ndevenish opened the pull request:
<!--
Thank you for your pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

Although tk [does a run_exports](https://github.com/conda-forge/tk-feedstock/blame/d9edffad87823da2e51a134b597291479ed02b66/recipe/meta.yaml#L31), it isn't [strong](https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#export-runtime-requirements), so having tk in build only doesn't propagate to runtime.

This causes a problem because this package installs `gitk`, which then fails because `tclsh` is missing.

ndevenish committed:
Make tk a host dependency, to pick up the run_exports

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found it was in an excellent condition.

ndevenish commented:
There doesn't seem to be a good way to test this, but `tk` is now being included in the conda-build test environments

jakirkham commented:
Thanks Nicholas! 🙏

Interestingly we use to have `tk` in `host`:

* https://github.com/conda-forge/git-feedstock/pull/42

Though for some reason it got moved:

* https://github.com/conda-forge/git-feedstock/pull/88

Am not actually seeing why that was

@isuruf do you remember?

timsnyder commented:
Looking at the details of #88 @isuruf didn't explain why he moved `tk` from `host` to `build`, he just said he was doing it in the commit message and there were no comments in the PR.  

My guesses are:
1.  would be that `tk` wasn't available on ARM OSX at that point and it was expedient to move it to get _most_ of the package to work correctly.  This is no longer the case, so maybe we can just put it back?
2. He was trying to make the git package smaller.  `tk` is only needed by `git-gui` and `gitk`.  If this is the reason, then moving those into split packages with dependence on `tk` might make sense.

@isuruf , I'm happy to do the more complicated PR if number 2 was the reason you moved it.  If it was just number 1, can we merge this PR after resolving conflicts?

isuruf commented:
Agree that `tk` should be in `host`.

dhirschfeld commented:
It looks like this is good to be merged.  @ndevenish - do you want to rebase so CI is green?

ndevenish committed:
MNT: Re-rendered with conda-smithy 3.60.0 and conda-forge-pinning 2026.04.16.17.17.57

Other tools:
- conda-build 26.3.0
- rattler-build 0.62.2
- rattler-build-conda-compat 1.4.13

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found some lint.

Here's what I've got...


For **recipe/meta.yaml**:

 * ❌ The MACOSX_DEPLOYMENT_TARGET key in /tmp/tmpqg7gpuha/cf_feedstock_ops_dir/recipe/conda_build_config.yaml needs to be removed or replaced by c_stdlib_version, appropriately restricted to osx
 * ❌ You are setting `c_stdlib_version` on osx below the current global baseline in conda-forge (11.0).


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/24992488049](https://github.com/conda-forge/conda-forge-webservices/actions/runs/24992488049). Examine the logs at this URL for more detail.</sub>

ndevenish commented:
Huh, why didn’t that rebase and conda-smithy work?

ndevenish committed:
MNT: Re-rendered with conda-smithy 3.61.2 and conda-forge-pinning 2026.04.25.10.12.57

Other tools:
- conda-build 26.3.0
- rattler-build 0.63.0
- rattler-build-conda-compat 1.4.14

ndevenish committed:
Update c_stdlib_version for the new OSX minimum

Additionally, remove MACOSX_DEPLOYMENT_TARGET as this is now supposed to
be controlled entirely from c_stdlib_version.

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found it was in an excellent condition.
